testhelpers/testhelpers.go

```diff
@@ -343,3 +349,40 @@ func MustReadFile(t *testing.T, path string) []byte {
 	}
 	return data
 }
+
+func ConditionallyBuildLifecycle(t *testing.T, dir string, goos string) {
+	AssertNil(t, os.MkdirAll(dir, 0755)) // Ensure directory exists
+
+	files, err := ioutil.ReadDir(dir)
+	AssertNil(t, err)
+
+	for _, file := range files {
+		if file.Name() == goos {
+			t.Log("WARNING: Not re-building lifecycle binaries, found existing binaries in", dir)
+			return
+		}
+	}
+
+	BuildLifecycle(t, dir, goos)
+}
+
+func BuildLifecycle(t *testing.T, dir string, goos string) {
+	cmd := exec.Command("make", "build-"+goos)
```

Comment:
Changed this from `runtime.GOOS`, but now that I'm looking it over again I'm not sure if that's what we want. I can change it back...

Reply:
Actually it seems that this function, where it was used previously (in `TestVersion`), was/is being called with `runtime.GOOS`, so this doesn't change any existing functionality.